rpcs3/util/types.hpp

```diff
@@ -1184,6 +1184,137 @@ namespace stx
 	};
 }
 
+// Read object of type T from raw pointer, array, string, vector, or any contiguous container
+template <typename T, typename U>
+constexpr T read_from_ptr(U&& array, usz pos = 0)
+{
+	// TODO: ensure array element types are trivial
+	static_assert(sizeof(T) % sizeof(array[0]) == 0);
+	std::decay_t<decltype(array[0])> buf[sizeof(T) / sizeof(array[0])];
+	if (!std::is_constant_evaluated())
+		std::memcpy(+buf, &array[pos], sizeof(buf));
+	else
+		for (usz i = 0; i < pos; buf[i] = array[pos + i], i++);
+	return std::bit_cast<T>(buf);
+}
+
+template <typename T, typename U>
+constexpr void write_to_ptr(const T& value, U&& array, usz pos = 0)
```

Comment:
Swapping argument places is more consistent with vm::write.

Reply:
It can have an overload which takes a single argument of array referece paired with index. So you can write wtite_to_ptr({array, 1}, data)

Reply:
That's right, fixed